packages/strapi-plugin-i18n/admin/src/hooks/useDefaultLocales/reducer.js

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable consistent-return */
+import produce from 'immer';
+
+export const initialState = {
+  defaultLocales: undefined,
+  isLoading: true,
+};
+
+const reducer = (state, action) =>
+  produce(state, draftState => {
+    switch (action.type) {
+      case 'GET_DATA': {
+        draftState.isLoading = true;
```

Comment:
you should also reset your state so you always return to the initialState:

`draftState = initialState`

Reply:
Not sure to mean what you mean?

Reply:
Well, when you refetch your data, you are starting a new cycle so instead of just playing with the `isLoading` you can directly return to the initial cycle which is `isLoading=false` & `defaultLocales=undefined`

Reply:
Ok, got it 😊